stdlib/public/core/CTypes.swift

```diff
@@ -101,6 +101,8 @@ public typealias CLongDouble = Double
 #endif
 #elseif os(OpenBSD)
 public typealias CLongDouble = Float80
+#elseif os(FreeBSD)
+public typealias CLongDouble = Float80
```

Comment:
This seems wrong.  FP80 is X86 only according to https://www.freebsd.org/cgi/man.cgi?query=arch&sektion=7&format=html.  This should be similar to what we have for Darwin.

Reply:
Good point, thanks!

I think that's also true for OpenBSD:

```
$ for os in freebsd openbsd; do
        for arch in i386 x86_64 armv7 arm64; do
                target="${arch}-${os}"                                                         
                echo -n "${target}: "                                                          
                clang-15 -target "${target}" -E -dM - </dev/null | fgrep __LDBL_MANT_DIG__
        done
        echo
done

i386-freebsd: #define __LDBL_MANT_DIG__ 64
x86_64-freebsd: #define __LDBL_MANT_DIG__ 64
armv7-freebsd: #define __LDBL_MANT_DIG__ 53
arm64-freebsd: #define __LDBL_MANT_DIG__ 113

i386-openbsd: #define __LDBL_MANT_DIG__ 64
x86_64-openbsd: #define __LDBL_MANT_DIG__ 64
armv7-openbsd: #define __LDBL_MANT_DIG__ 53
arm64-openbsd: #define __LDBL_MANT_DIG__ 113
```

I've added an x86_64 || i386 conditional to the FreeBSD branch.  @compnerd & @3405691582, would you like me to do the same for OpenBSD, or would you prefer I leave it alone (or just add a comment)?

Reply:
A separate PR for fixing O/BSD would be appreciated!

Reply:
Currently it is an error to attempt to build Swift on anything but amd64 right now, so it's not strictly necessary to fix for this platform right now. It might be worthwhile doing, but it might be misleading if you see the conditional and yet get a cmake error trying to bring up the platform on the architeture though.